### PR TITLE
Dev 0.3.3

### PR DESCRIPTION
**Pull request by enmerk4r**
Compatibility improvements with JSwan

<h3>Greptile Summary</h3>

This PR adds Newtonsoft.Json interoperability to all four Grasshopper Goo types (`JsonObjectGoo`, `JsonArrayGoo`, `JsonValueGoo`, `JsonNodeGoo`) so that Swiftlet JSON outputs can be consumed directly by components that expect Newtonsoft `JToken` values — most notably JSwan.

- A new `JsonNewtonsoftInterop` class handles round-trip conversion between `System.Text.Json.Nodes` and Newtonsoft `JToken` via JSON serialization, and a `JsonGooDynamicMetaObject` rewrites dynamic `.Value` member access to call `ScriptVariable()`, returning a `JToken` instead of the underlying `JsonNode`.
- All four Goo types now implement `IDynamicMetaObjectProvider`, override `ScriptVariable()` to return the equivalent Newtonsoft token, and add `CastTo`/`CastFrom` branches for `JObject`, `JArray`, `JValue`, `JToken`, and `object` target types.
- Integration tests cover both directions of the cast for all Goo types, including null semantics and the `dynamic.Value` path; core tests are upgraded from substring `Contains` checks to structural JSON equality assertions.

<h3>Confidence Score: 5/5</h3>

Safe to merge; the interop logic is well-encapsulated and the two new integration tests exercise all four Goo types in both cast directions including null semantics.

The conversion paths are straightforward (serialize to JSON string, re-parse), the DynamicMetaObject restriction is correctly scoped per concrete type, and the test coverage is thorough. The only new observations are redundant fall-through paths in CastFrom that are harmless in practice.

The CastFrom implementations in JsonObjectGoo, JsonArrayGoo, and JsonValueGoo all have the redundant JToken fall-through pattern worth tidying up.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| src/Swiftlet.Gh.Rhino8/JsonNewtonsoftInterop.cs | New file providing bidirectional conversion between System.Text.Json.Nodes and Newtonsoft JToken, plus a DynamicMetaObject that rewrites .Value member access to call ScriptVariable(). The JRaw/null edge case in FromJToken was flagged in a prior thread. |
| src/Swiftlet.Gh.Rhino8/Goo/JsonObjectGoo.cs | Adds IDynamicMetaObjectProvider, ScriptVariable() returning JObject, and bidirectional JToken cast support. The JToken fallback block is redundant for JObject sources. |
| src/Swiftlet.Gh.Rhino8/Goo/JsonArrayGoo.cs | Adds IDynamicMetaObjectProvider, ScriptVariable() returning JArray, and bidirectional JToken cast support; contains the same redundant JToken fallback block pattern as JsonObjectGoo. |
| src/Swiftlet.Gh.Rhino8/Goo/JsonValueGoo.cs | Adds IDynamicMetaObjectProvider, ScriptVariable() returning JValue/JValue.CreateNull(), and bidirectional JToken cast support; the JValue-specific block in CastFrom silently falls through to the JToken block on the rare path where neither null nor JsonValue is matched. |
| src/Swiftlet.Gh.Rhino8/Goo/JsonNodeGoo.cs | Adds IDynamicMetaObjectProvider, ScriptVariable() returning a JToken, and a JToken case in CastFrom that always returns true (the null-inconsistency edge case was flagged in a prior thread). |
| tests/Swiftlet.Integration.Tests/Program.cs | Adds two well-structured integration tests covering round-trip cast to/from Newtonsoft tokens for all four Goo types, including null semantics and the dynamic .Value accessor path. |
| tests/Swiftlet.Core.Tests/Program.cs | Upgrades McpClientConfigBuilder tests from substring Contains assertions to full structural JSON equality, improving assertion quality. |
| build/packaging/Publish-Target.ps1 | Adds Newtonsoft.Json.dll to the Yak runtime dependency list so it is bundled alongside the plugin assembly. |
| build/packaging/yak-manifest-template.yml | Adds the JSwan package GUID as a Yak keyword (cross-package compatibility declaration) and sets platform: any. |

</details>

<h3>Flowchart</h3>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[GH Component Output\nJsonObjectGoo / JsonArrayGoo\nJsonValueGoo / JsonNodeGoo] --> B{Target context?}

    B -->|Grasshopper wire\nCastTo JsonNodeGoo| C[JsonNodeGoo\nSystem.Text.Json.Nodes]
    B -->|CastTo JObject / JArray / JValue| D[JsonNewtonsoftInterop.ToJToken\nnode.ToJsonString → JToken.Parse]
    B -->|CastTo object or ScriptVariable| D
    B -->|dynamic .Value| E[JsonGooDynamicMetaObject\nBindGetMember → ScriptVariable]

    D --> F{Token type}
    F -->|JObject| G[Newtonsoft JObject]
    F -->|JArray| H[Newtonsoft JArray]
    F -->|null / RepresentsJsonNull| I[JValue.CreateNull]
    F -->|scalar| J[Newtonsoft JValue]

    E --> D

    K[External JToken\ne.g. from JSwan] --> L[CastFrom]
    L --> M[JsonNewtonsoftInterop.FromJToken\nJToken.ToString → JsonNode.Parse]
    M --> N[System.Text.Json.Nodes\nstored in Goo.Value]
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
flowchart TD
    A[GH Component Output\nJsonObjectGoo / JsonArrayGoo\nJsonValueGoo / JsonNodeGoo] --> B{Target context?}

    B -->|Grasshopper wire\nCastTo JsonNodeGoo| C[JsonNodeGoo\nSystem.Text.Json.Nodes]
    B -->|CastTo JObject / JArray / JValue| D[JsonNewtonsoftInterop.ToJToken\nnode.ToJsonString → JToken.Parse]
    B -->|CastTo object or ScriptVariable| D
    B -->|dynamic .Value| E[JsonGooDynamicMetaObject\nBindGetMember → ScriptVariable]

    D --> F{Token type}
    F -->|JObject| G[Newtonsoft JObject]
    F -->|JArray| H[Newtonsoft JArray]
    F -->|null / RepresentsJsonNull| I[JValue.CreateNull]
    F -->|scalar| J[Newtonsoft JValue]

    E --> D

    K[External JToken\ne.g. from JSwan] --> L[CastFrom]
    L --> M[JsonNewtonsoftInterop.FromJToken\nJToken.ToString → JsonNode.Parse]
    M --> N[System.Text.Json.Nodes\nstored in Goo.Value]
```

</a>

<sub>Reviews (2): Last reviewed commit: ["Update src/Swiftlet.Gh.Rhino8/JsonNewton..."](https://github.com/enmerk4r/swiftlet/commit/d61274f1139db1498a488c8746d41eb4a1d4d355) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=39066937)</sub>

### CODE REVIEW

File: src/Swiftlet.Gh.Rhino8/Goo/JsonObjectGoo.cs [hunk 1]
```diff
@@ -23,15 +26,46 @@ public JsonObjectGoo(JsonObject? obj)
 
     public override IGH_Goo Duplicate() => new JsonObjectGoo(Value);
 
+    public override object ScriptVariable()
+    {
+        return JsonNewtonsoftInterop.ToJToken(Value);
+    }
```
[the remark this answers is not in the document]
Comment (enmerk4r) [reply src/Swiftlet.Gh.Rhino8/Goo/JsonObjectGoo.cs:32]: This is fine, since all older versions before 0.3.x relied on Newtonsoft. So, if anything, this will actually improve our compatibility with old definitions.
